### ISSUE TITLE
ExecuteCommandTask should validate configured events

### ISSUE THREAD

**Issue by RonRademaker** [accompli/accompli#214]
To make sure opcache is cleared I restart Apache after a deployment using this task:

``` json
{
    "class": "Accompli\\Task\\ExecuteCommandTask",
    "events": ["accompli.deploy_release_complete"],
    "command": "sudo /etc/init.d/apache2 restart"
}
```

The user as which accompli runs remotely has permissions to restart apache using sudo without a password prompt.

However, the ExecuteCommandTask can't be used with the `AccompliEvents::DEPLOY_RELEASE_COMPLETE` event, only with `AccompliEvents::PREPARE_RELEASE`, `AccompliEvents::INSTALL_RELEASE`, `AccompliEvents::DEPLOY_RELEASE` and `AccompliEvents::ROLLBACK_RELEASE`. Accompli simply doesn't perform the task without any error message. I think it should either support all events, or trigger some kind of error to let the user know why the task is not executed.

**Comment by niels-nijens** [comment]
The event cycle ends with either an `AccompliEvents::DEPLOY_RELEASE_COMPLETE` or an `AccompliEvents::DEPLOY_RELEASE_FAILED`. Your example is actually part of the deployment as failure of it will leave a broken deployment, so it can't be executed when Accompli has actually marked the deployment as complete.

I agree with you that the `ExecuteCommandTask` should validate the configured events during construction of the task instance.

**Comment by RonRademaker** [comment]
> Your example is actually part of the deployment as failure of it will leave a broken deployment

You are correct. I considered attaching to `AccompliEvents::DEPLOY_RELEASE`, however the (preconfigured) priority in `Accompli\\Task\\ExecuteCommandTask` is 10, where the symlink is updated with priority 0. That would mean the restart would be done before the symlink is updated, which is incorrect.
